sdk/core/azure-core/inc/azure/core/http/websockets/curl_websockets_transport.hpp

```diff
@@ -15,7 +15,7 @@
 #include "azure/core/http/websockets/websockets_transport.hpp"
 #include <memory>
 
-namespace Azure { namespace Core { namespace Http { namespace WebSockets {
+namespace Azure { namespace Core { namespace Http { namespace _detail { namespace WebSockets {
```

Comment:
Do we have any precedence for `_detail` to be in the middle, rather than at the end?

We previously decided to always have `_detail` be the last "node" in the namespaces list. It makes the namespace structure simpler and easy to author (fewer decisions).
I realize a downside of that is we'll end up having some sub-namespaces "public", with nothing in them that are actually public, at the moment. But I think that's OK since it is more future proof/extensible.

From https://github.com/Azure/azure-sdk-for-cpp/issues/1789:
> Each package can define _detail namespace(s) as they see fit under one condition. **The _detail namespace must be the last namespace in the namespace chain.**

Reply:
_detail was Rick's suggestion to make it clear that service authors should not take dependencies on this type. I'm open to other suggestions however (see the separate email thread).

Reply:
Having everything in `_detail` makes perfect sense to me. The suggestion is to just have that be at the end instead.

`namespace Azure { namespace Core { namespace Http { namespace WebSockets { namespace _detail {`